### PR TITLE
Fix autostart sleep (again)

### PR DESCRIPTION
**Pull request by Firebie**
Please try and fill out this template where possible, not all fields are required and can be removed.

* **Please check if the PR fulfills these requirements**
- [x] The commit message makes sense
- [x] The changes have been tested locally
- [] Are there any breaking changes

* **What kind of change does this PR introduce?**
(Bug fix, feature, docs update, ...)
Fix for https://github.com/Ralim/ts100/issues/571

* **What is the current behavior?**
(You can also link to an open issue here)
Fix for https://github.com/Ralim/ts100/issues/571

* **What is the new behavior (if this is a feature change)?**
Fix for https://github.com/Ralim/ts100/issues/571

* **Does this PR introduce a breaking change?**
(What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by Firebie** [comment]
Updated to latest discussion in https://github.com/Ralim/ts100/issues/571

**Comment by Ralim** [comment]
Please see previous review comments before this can be merged.

What would be ideal is to actually fix the flaw in the accelerometer algorithm, but most of this is fine otherwise.

Note that sleep is at sleep temperature. If you want to have a mode where it starts "off" and then wakes up, this should have a new menu item to enable (in autostart menu)

**Comment by Firebie** [comment]
Sorry, but I can't find where I can see comments.

**Comment by Firebie** [comment]
> Note that sleep is at sleep temperature. If you want to have a mode where it starts "off" and then wakes up, this should have a new menu item to enable (in autostart menu)
May be another letter to F/T/S - O?

**Comment by Ralim** [comment]
Yeah adding another combination, it requires adding it to all the translation files but it's not too hard to add

**Comment by Firebie** [comment]
We anyway need (xTaskGetTickCount() > 100) as initialization can be fast and following conditions will be true without read accel move/button press.

File: workspace/TS100/Core/Src/GUIThread.cpp [hunk 1]
```diff
@@ -352,27 +352,28 @@ static void gui_solderingTempAdjust() {
...
 #endif
 		if (systemSettings.temperatureInF) {
-			currentTempTargetDegC = TipThermoModel::convertFtoC(
+			currentTempTargetDegC = jumpToSleep ? 0 : TipThermoModel::convertFtoC(
 					min(systemSettings.SleepTemp,
 							systemSettings.SolderingTemp));
 		} else {
-			currentTempTargetDegC = (min(systemSettings.SleepTemp,
-					systemSettings.SolderingTemp));
+			currentTempTargetDegC = jumpToSleep ? 0 : min(systemSettings.SleepTemp,
```
[the remark this answers is not in the document]
Comment (Firebie) [reply workspace/TS100/Core/Src/GUIThread.cpp]: Fixed in latest commit

File: workspace/TS100/Core/Src/GUIThread.cpp [hunk 2]
```diff
@@ -352,27 +352,28 @@ static void gui_solderingTempAdjust() {
...
 
 	for (;;) {
 		ButtonState buttons = getButtonState();
 		if (buttons)
 			return 0;
-		if ((lastMovementTime > 100 && (xTaskGetTickCount() - lastMovementTime < 100))
-				|| (xTaskGetTickCount() - lastButtonTime < 100))
+		if (xTaskGetTickCount() > 200
+			&& ((xTaskGetTickCount() - lastMovementTime < 100)
+				|| (xTaskGetTickCount() - lastButtonTime < 100)))
```
[the remark this answers is not in the document]
Comment (Firebie) [reply workspace/TS100/Core/Src/GUIThread.cpp]: Fixed in latest commit

File: workspace/TS100/Core/Src/GUIThread.cpp [hunk 3]
```diff
@@ -352,27 +353,28 @@ static void gui_solderingTempAdjust() {
 	}
 }
 
-static int gui_SolderingSleepingMode() {
+static int gui_SolderingSleepingMode(uint8_t stayOff) {
```
Comment (Ralim) [inline workspace/TS100/Core/Src/GUIThread.cpp]: Why is this not a `bool` ?
its the same memory footprint, but `boot` is more clear as to its intent.

**Comment by Firebie** [comment]
Done

**Pull request merged**